liquibase-core/src/main/java/liquibase/exception/MigrationFailedException.java

```diff
@@ -14,17 +19,40 @@ public MigrationFailedException() {
     public MigrationFailedException(ChangeSet failedChangeSet, String message) {
         super(message);
         this.failedChangeSetName = failedChangeSet.toString(false);
+        this.failedChangeSets = filterFailedChangeSets(failedChangeSet);
     }
 
 
     public MigrationFailedException(ChangeSet failedChangeSet, String message, Throwable cause) {
         super(message, cause);
         this.failedChangeSetName = failedChangeSet.toString(false);
+        this.failedChangeSets = filterFailedChangeSets(failedChangeSet);
     }
 
     public MigrationFailedException(ChangeSet failedChangeSet, Throwable cause) {
         super(cause);
         this.failedChangeSetName = failedChangeSet.toString(false);
+        this.failedChangeSets = filterFailedChangeSets(failedChangeSet);
+    }
+
+    public boolean containsFailedChangeset() {
+        return !(failedChangeSets == null || failedChangeSets.isEmpty());
+    }
+
+    public List<ChangeSet> getFailedChangeSets() {
+        return new LinkedList<>(failedChangeSets);
```

Comment:
Any reason you're creating a new `LinkedList` object here? I would think you could return `failedChangeSets` directly.

Reply:
I was thinking of preventing devs from mutating the object outside of the class. I would've used `Collections.unmodifiableList` but it prevents us from reversing the list in the function `rollbackChangeSets` down below.